Wikipedia/Code/MWKList.h

```diff
@@ -40,9 +40,9 @@ typedef id<NSCopying, NSObject> MWKListIndex;
 
 - (EntryType)entryAtIndex:(NSUInteger)index;
 
-- (EntryType __nullable)entryForListIndex:(IndexType)listIndex;
+- (EntryType __nullable)entryForListIndex:(MWKListIndex)listIndex;
```

Comment:
this effectively removes the generic aspect of the `MWKList` API, meaning all the entry-related APIs won't auto-complete to the type parameters for the specific subclass.  AFAICT, the warning here is in `MWKHistoryList.addEntry`, which should be using `entry.title`, not `entry.listIndex`.